pkg/controller/migration/convert/network.go

```diff
@@ -119,6 +130,107 @@ func handleNetwork(c *components, install *Installation) error {
 	return nil
 }
 
+func handleNonCalicoCNI(c *components, install *Installation) error {
+	plugin, err := getCNIPlugin(c)
+	if err != nil {
+		return err
+	}
+	if plugin == operatorv1.PluginCalico {
+		return nil
+	}
+
+	context := fmt.Sprintf("detected %s CNI plugin", plugin)
+
+	icc := getContainer(c.node.Spec.Template.Spec, containerInstallCNI)
+	if icc != nil {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("%s, %s container is not supported in the node daemonset", context, containerInstallCNI),
+		}
+	}
+
+	// CALICO_NETWORKING_BACKEND
+	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return err
+	}
+	if netBackend == nil || *netBackend != "none" {
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, CALICO_NETWORKING_BACKEND=none is expected", context)}
+	}
+
+	if install.Spec.CNI == nil {
+		install.Spec.CNI = &operatorv1.CNISpec{}
+	}
+
+	switch plugin {
+	case operatorv1.PluginAmazonVPC:
+		install.Spec.CNI.Type = plugin
+		// FELIX_IPTABLESMANGLEALLOWACTION
```

Comment:
This comment isn't super useful - I'd recommend removing it or flushing it out to explain why we care about this field (probably the latter)